src/luarocks/fetch.lua

```diff
@@ -244,21 +271,46 @@ function fetch.load_local_rockspec(filename, quick)
                       or rockspec.source.module
                       or ((filebase:match("%.lua$") or filebase:match("%.c$")) and ".")
                       or base
-   if rockspec.dependencies then
-      for i = 1, #rockspec.dependencies do
-         local parsed, err = deps.parse_dep(rockspec.dependencies[i])
-         if not parsed then
-            return nil, "Parse error processing dependency '"..rockspec.dependencies[i].."': "..tostring(err)
-         end
-         rockspec.dependencies[i] = parsed
-      end
-   else
-      rockspec.dependencies = {}
+   local ok, err = parse_deps(rockspec, "build_dependencies")
+   if not ok then
+      return nil, err
+   end
+   local ok, err = parse_deps(rockspec, "dependencies")
+   if not ok then
+      return nil, err
    end
    if not quick then
       path.configure_paths(rockspec)
    end
 
+   -- Fulfill build_dependencies
+   -- XXX ignores depmode from command line
+   local ok, err = deps.fulfill_dependencies(rockspec, cfg.deps_mode, true)
+   if not ok then
+      return nil, err
+   end
+
+   -- Load addons
+   addon.reset()
+   if rockspec.addons then
+      for _, name in ipairs(rockspec.addons) do
+         local ok, err = addon.load(name)
+         if not ok then
+            return nil, filename..": cannot load addon "..addon..": "..err
```

Comment:
should be `return nil, filename..": cannot load addon "..name..": "..err` — got a crash there